### PR TITLE
DOC-1255 Update GDK and plugin terminology

### PR DESCRIPTION
**Pull request by anne-edwards**
Primary reviewer: @ElleEss 

**What I have done:**

I updated the wording around the GDK/plugin for:
- the readme
- the versioning page
- the tech overview pages, including the networking switch diagram

I also changed "Fork" to "fork" (lower case) throughout, and checked that "plugin" is lower case already.

<br>

**What I haven't done:**
- home page  
- Starter Template
- Example Project
- get started
<br>

**Related updates to readmes**
- [plugin repo](https://github.com/spatialos/UnrealGDK/pull/1287)
- [Example Project repo](https://github.com/spatialos/UnrealGDKExampleProject/pull/43)
- [fork repo](https://github.com/improbableio/UnrealEngine/pull/242)

### CODE REVIEW

**Comment by anne-edwards** [comment]
@ElleEss OK, I think this is now ready for review!

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # The SpatialOS Game Development Kit for Unreal (alpha)
```
Comment (anne-edwards) [inline README.md]: I might be being dim, but I can't find where this file gets displayed. It's quite different from the one [here](https://github.com/spatialos/UnrealGDK/blob/release/README.md) and it doesn't appear to be in the docs either.

Comment (anne-edwards) [reply README.md]: AAH, I've just changed branch to `docs-release` and it's in there, but it was last edited in April. I think this probably means the `release` branch readme is more accurate and up-to-date, so I guess I need to make changes to that by branching from the main `release` branch?

Comment (ElleEss) [reply README.md]: I am enjoying this one-person conversation. It's very familiar to me.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
-The SpatialOS Game Development Kit (GDK) for Unreal is an Unreal Engine 4 (UE4) [plugin (Unreal documentation)](https://docs.unrealengine.com/en-us/Programming/Plugins) which allows you to host your game and combine multiple dedicated server instances across one seamless game world, whilst using the Unreal Engine networking API.
+The SpatialOS Game Development Kit (GDK) for Unreal is an Unreal Engine fork and plugin with associated projects. It enables you to host your game and combine multiple dedicated server instances across one seamless game world, while using the Unreal Engine networking API.
```
[the remark this answers is not in the document]
Comment (anne-edwards) [reply README.md]: Yes!

**Pull request merged**